Library/Homebrew/cask/lib/hbc/exceptions.rb

```diff
@@ -3,56 +3,40 @@ class CaskError < RuntimeError; end
 
   class AbstractCaskErrorWithToken < CaskError
     attr_reader :token
+    attr_reader :reason
 
-    def initialize(token)
+    def initialize(token, reason = nil)
       @token = token
+      @reason = reason.to_s
     end
   end
 
   class CaskNotInstalledError < AbstractCaskErrorWithToken
     def to_s
-      "#{token} is not installed"
+      "Cask '#{token}' is not installed."
     end
   end
 
   class CaskUnavailableError < AbstractCaskErrorWithToken
     def to_s
-      "No available Cask for #{token}"
+      "Cask '#{token}' is unavailable" << (reason.empty? ? "." : ": #{reason}")
```

Comment:
Nitpick: should this be `": #{reason}."` for consistency, so all error messages end with a "."?